### ISSUE TITLE
[Bug]: Segmentation fault on fresh project

### ISSUE THREAD

**Issue by Munksgaard** [shuttle-hq/shuttle#694]
### What happened?

I created a new project using `cargo shuttle init` (v0.11.2), moved into the directory and ran `cargo shuttle run`. After a bit of compilation I get the following:

```console
    [...]
    Finished dev [unoptimized + debuginfo] target(s) in 54.86s

Starting <my-project> on http://127.0.0.1:8000
Segmentation fault (core dumped)
```

### Version

v0.11.2

### Which operating systems are you seeing the problem on?

Linux

### Which CPU architectures are you seeing the problem on?

x86_64

### Duplicate declaration

- [X] I have searched the issues and there are none like this.

**Comment by Munksgaard** [comment]
It was a rocket project, by the way.

**Comment by oddgrd** [comment]
Hey! We see this happen spuriously on linux, and more regularly on macOS. We currently compile users projects to dynamic libraries (`.so` files), and we need an unsafe FFI to load them. Since this has been rare on linux, I don't have a good suggestion for how you can remedy this right now. 

The good news is we're refactoring our codegen and how we compile users projects this week, which means we'll be getting rid of the FFI early next week. This should be a thing of the past then.

**Comment by Munksgaard** [comment]
Thanks for the response. I'm on NixOS, if that makes a difference. I'll wait for your refactoring and see if that fixes the problem.

**Comment by oddgrd** [comment]
This should be resolved after 0.12, but please re-open this if you still experience it. Thanks!
